### PR TITLE
Improve pyEnum.

### PR DESCRIPTION
**Pull request by Hoikas**
The following issues were corrected:
- Fixed a ~hypothetical~ memory leak when adding the enum object to the module fails. (Apparently this leak is not so hypothetical. VLD says this fixes many leaks.)
- Fixed the failure return for EnumValue_hash to actually indicate failure.
- Avoided copying an std::map.
- Replaced an std::map used only for iteration with a vector of tuples.
- Fixed the code style such that I no longer want to stab my eyes out.

Sorry about the noise from all the semicolon removals 😞

### CODE REVIEW

File: Sources/Plasma/FeatureLib/pfPython/pyEnum.cpp [hunk 1]
```diff
@@ -50,212 +50,151 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
...
 
+#include "pyEnum.h"
+#include "pyGlueHelpers.h"
+#include "pyObjectRef.h"
+
 struct EnumValue {
     PyObject_HEAD
-    long value;
-    char* name;
+    hsSsize_t value{};
```
Comment (zrax) [inline Sources/Plasma/FeatureLib/pfPython/pyEnum.cpp:62]: `hsSsize_t` and `Py_ssize_t` are not guaranteed to be the same type...  You should use `Py_ssize_t` (or another type documented/supported by Python) to better guarantee compatibility with the Python API.

File: Sources/Plasma/FeatureLib/pfPython/pyEnum.cpp [hunk 2]
```diff
@@ -360,21 +299,21 @@ PYTHON_TYPE_START(EnumValue)
     nullptr,                            /* tp_getattr */
     nullptr,                            /* tp_setattr */
     nullptr,                            /* tp_as_async */
-    EnumValue_repr,                     /* tp_repr */
+    (reprfunc)EnumValue_repr,           /* tp_repr */
```
Comment (zrax) [inline Sources/Plasma/FeatureLib/pfPython/pyEnum.cpp:302]: IIRC even with a cast, this will trigger a warning on newer versions of GCC...

Comment (Hoikas) [reply Sources/Plasma/FeatureLib/pfPython/pyEnum.cpp:302]: Thanks GCC. So much for avoiding dumb casts 😞

File: Sources/Plasma/FeatureLib/pfPython/pyGlueHelpers.h [hunk 3]
```diff
@@ -611,12 +611,12 @@ static PyObject *methodName(PyObject *self) /* and now for the actual function *
 /////////////////////////////////////////////////////////////////////
 
 // the start of an enum block
-#define PYTHON_ENUM_START(enumName) std::map<std::string, int> enumName##_enumValues
+#define PYTHON_ENUM_START(enumName) std::vector<std::tuple<ST::string, hsSsize_t>> enumName##_enumValues{
```
Comment (zrax) [inline Sources/Plasma/FeatureLib/pfPython/pyGlueHelpers.h:614]: Missed one...

File: Sources/Plasma/FeatureLib/pfPython/pyEnum.cpp [hunk 4]
```diff
@@ -366,15 +304,15 @@ PYTHON_TYPE_START(EnumValue)
...
+    nullptr,                            /* tp_str */
     nullptr,                            /* tp_getattro */
     nullptr,                            /* tp_setattro */
     nullptr,                            /* tp_as_buffer */
     Py_TPFLAGS_DEFAULT
     | Py_TPFLAGS_BASETYPE,              /* tp_flags */
     "A basic enumeration value",        /* tp_doc */
     nullptr,                            /* tp_traverse */
-    nullptr,                            /* tp_clear */
+    (inquiry)EnumValue_clear,           /* tp_clear */
```
Comment (zrax) [inline Sources/Plasma/FeatureLib/pfPython/pyEnum.cpp:315]: This cast should probably be fixed too

**Review by zrax — APPROVED** [review]

**Pull request merged**